solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/health/handlers/SolaceFlowHealthEventHandler.java

```diff
@@ -0,0 +1,71 @@
+package com.solace.spring.cloud.stream.binder.health.handlers;
+
+import com.solace.spring.cloud.stream.binder.health.indicators.FlowHealthIndicator;
+import com.solace.spring.cloud.stream.binder.util.XMLMessageMapper;
+import com.solacesystems.jcsmp.FlowEvent;
+import com.solacesystems.jcsmp.FlowEventArgs;
+import com.solacesystems.jcsmp.FlowEventHandler;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+public class SolaceFlowHealthEventHandler implements FlowEventHandler {
+	private final FlowHealthIndicator flowHealthIndicator;
+	private final XMLMessageMapper xmlMessageMapper;
+	private final String flowReceiverContainerId;
+	private static final Log logger = LogFactory.getLog(SolaceFlowHealthEventHandler.class);
+
+	public SolaceFlowHealthEventHandler(XMLMessageMapper xmlMessageMapper,
+	                                    String flowReceiverContainerId,
+	                                    FlowHealthIndicator flowHealthIndicator) {
+		this.flowHealthIndicator = flowHealthIndicator;
+		this.xmlMessageMapper = xmlMessageMapper;
+		this.flowReceiverContainerId = flowReceiverContainerId;
+	}
+
+	@Override
+	public void handleEvent(Object source, FlowEventArgs flowEventArgs) {
+		if (logger.isDebugEnabled()) {
+			logger.debug(String.format("(%s): Received Solace Flow event [%s].", source, flowEventArgs));
+		}
+		if (flowEventArgs.getEvent() != null) {
+			switch (flowEventArgs.getEvent()) {
+				case FLOW_DOWN:
+					flowHealthIndicator.down(flowEventArgs);
+					break;
+				case FLOW_RECONNECTING:
+					resetXmlMessageMapperProperties(flowEventArgs);
+					flowHealthIndicator.reconnecting(flowEventArgs);
+					break;
+				case FLOW_UP:
+				case FLOW_RECONNECTED:
+					flowHealthIndicator.up();
+					break;
+			}
+		}
+	}
+
+	private void resetXmlMessageMapperProperties(FlowEventArgs flowEventArgs) {
+		if(xmlMessageMapper != null){
+			if (logger.isDebugEnabled()) {
+				logger.debug(String.format("Received flow event %s for flow receiver container %s. Will clear ignored properties.",
+						flowEventArgs.getEvent().name(), flowReceiverContainerId));
+			}
+			xmlMessageMapper.resetIgnoredProperties(flowReceiverContainerId);
+		}
+	}
+
+	/**
+	 * This method is used only on during  event of binding to source
+	 */
+	public void setHealthStatusUp() {
+		this.flowHealthIndicator.up();
+	}
+
+	/**
+	 * On the contrary, this one is used only when source is unbind
+	 */
+	public void setHealthStatusDown() {
+		FlowEventArgs flowEventArgs = new FlowEventArgs(FlowEvent.FLOW_DOWN,"Flow receiver has been unbind", null, 201);
+		this.flowHealthIndicator.down(flowEventArgs);
+	}
```

Comment:
Instead of being marked as `DOWN`, flow health indicators need to be deleted when a binding unbinds. Since the underlying resource (i.e. the flow receiver) is terminally closed when unbinds happen.

Reply:
Setting health `DOWN` (might be we need to use `DISCONNECTED` though or some other custom status), is intentional, as it may not work with other products well enough.

As an example, Connector Manager always show the health of underlying connection in Workflows view. At the moment there is a health status for the whole connection, because of lacking this feature, however after releasing this one, it will catch up aggregated status for every for every binding separately.

In case of removing health indicators completely, there will be nothing to set as a status in a view and thus we need to provide current status there as well, which as I said might be custom, like `DOWN`, `DISCONNECTED` or `UNBIND`.

Reply:
After a careful consideration of all possible options we will proceed with deleting Health Indicators on any `unbind` events.